synapse/server.py

```diff
@@ -225,6 +227,7 @@ def __init__(self, hostname, reactor=None, **kwargs):
         self.registration_ratelimiter = Ratelimiter()
 
         self.datastore = None
```

Comment:
should we make this a local variable now rather than have it duplicate `datastores.main` ?